compiler/env/OMRObjectModel.hpp

```diff
@@ -110,12 +111,43 @@ class ObjectModel
    /**
    * @brief: Returns true if concurrent scavenging enabled in the VM's GC
    */
-   bool shouldGenerateReadBarriersForFieldLoads() { return false; };
+   bool shouldGenerateReadBarriersForFieldLoads() { return false; }
 
    /**
    * @brief: Returns true if option for software read barriers is enabled in the VM's GC
    */
-   bool shouldReplaceGuardedLoadWithSoftwareReadBarrier() { return false; };
+   bool shouldReplaceGuardedLoadWithSoftwareReadBarrier() { return false; }
+
+   /**
+    * \brief Dereference a static reference field
+    *
+    * \param
+    *     comp TR::Compilation
+    *
+    * \param
+    *     fieldAddress Address of field to be dereferenced
+    *
+    * \param
+    *     symRef symref of the node representing the field dereference
+    *
+    * \return
+    *     dereferenced field
+    */
+   uintptrj_t getStaticReferenceFieldAtAddress(TR::Compilation* comp, uintptrj_t fieldAddress, TR::SymbolReference *symRef) { return *(uintptrj_t *)fieldAddress; }
```

Comment:
I would vouch to say this should assert via `notImplemented` rather than providing a default implementation. How you read static fields seems very downstream-project specific (similarly for below).

Reply:
I'll have to check if there's any code not guarded by the `J9_PROJECT_SPECIFIC` macro that makes this call; if there is then we can't add that assert. There is also the counter proposal that the default behaviour (considering that we're only adding this API to account for read barriers of collected references, and not for things like the `TR_OpaqueClassBlock`) should be just a simple dereference.

Reply:
Looks like every call to `getStaticReferenceFieldAtAddress` is protected by `J9_PROJECT_SPECIFIC` so I'll update the OMR implementation to require a downstream implementation.